src/daemon/collectd.c

```diff
@@ -71,7 +71,8 @@ static void sig_usr1_handler(int __attribute__((unused)) signal) {
    * so it should be done asynchronously */
   pthread_attr_init(&attr);
   pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);
-  pthread_create(&thread, &attr, do_flush, NULL);
+  if (pthread_create(&thread, &attr, do_flush, NULL) != 0)
+    fprintf(stderr, "Can not create thread: %s.\n", strerror(errno));
```

Comment:
At this point we're already forked so stderr is closed.
You're also using strerror() which is not async signal safe.

I don't think handling the error buys us anything here.